server/odc-service/src/main/java/com/oceanbase/odc/service/connection/table/TableService.java

```diff
@@ -98,30 +103,71 @@ public class TableService {
     @Transactional(rollbackFor = Exception.class)
     @SkipAuthorize("permission check inside")
     public List<Table> list(@NonNull @Valid QueryTableParams params) throws SQLException, InterruptedException {
+        List<DBObjectType> types = params.getTypes();
         Database database = databaseService.detail(params.getDatabaseId());
         ConnectionConfig dataSource = database.getDataSource();
         OBConsoleDataSourceFactory factory = new OBConsoleDataSourceFactory(dataSource, true);
+        List<Table> tables = new ArrayList<>();
         try (SingleConnectionDataSource ds = (SingleConnectionDataSource) factory.getDataSource();
                 Connection conn = ds.getConnection()) {
-            TableExtensionPoint point = SchemaPluginUtil.getTableExtension(dataSource.getDialectType());
-            Set<String> latestTableNames = point.list(conn, database.getName())
-                    .stream().map(DBObjectIdentity::getName).collect(Collectors.toSet());
-            if (authenticationFacade.currentUser().getOrganizationType() == OrganizationType.INDIVIDUAL) {
-                return latestTableNames.stream().map(tableName -> {
-                    Table table = new Table();
-                    table.setName(tableName);
-                    table.setAuthorizedPermissionTypes(new HashSet<>(DatabasePermissionType.all()));
-                    return table;
-                }).collect(Collectors.toList());
+            if (types == null || types.contains(DBObjectType.TABLE)) {
+                // add logic about table and make table.type==DBObjectType.TABLE
+                TableExtensionPoint point = SchemaPluginUtil.getTableExtension(dataSource.getDialectType());
```

Comment:
TableExtensionPoint point; `point` variable may rename to `tableExtention`

and what if get extension failed or get null ?
often we can juedge if feature is supported via extesion point exists or not.